### PR TITLE
LP-1959 Idea listing functionality

### PR DESCRIPTION
**Pull request by abdullah998**
### Description

[LP-1959](https://philanthropyu.atlassian.net/browse/LP-1959)

### CODE REVIEW

File: openedx/features/idea/helpers.py [hunk 1]
```diff
@@ -12,3 +12,46 @@ def upload_to_path(instance, filename, folder):
...
     )
+
+
+def pretty_date(time):
+    """
+    Get a datetime object and return a pretty string like 'an hour ago',
+    'Yesterday', '3 months ago', 'just now', etc
+    """
+    from datetime import datetime
+    from django.utils.timezone import utc
```
[the remark this answers is not in the document]
Comment (abdullah998) [reply openedx/features/idea/helpers.py]: Made them global

File: openedx/features/idea/helpers.py [hunk 2]
```diff
@@ -12,3 +12,46 @@ def upload_to_path(instance, filename, folder):
         filename=filename,
         folder=folder
     )
+
+
+def pretty_date(time):
+    """
+    Get a datetime object and return a pretty string like 'an hour ago',
+    'Yesterday', '3 months ago', 'just now', etc
+    """
```
[the remark this answers is not in the document]
Comment (abdullah998) [reply openedx/features/idea/helpers.py:25]: Changed it Usman bhai

File: openedx/features/idea/helpers.py [hunk 3]
```diff
@@ -12,3 +16,43 @@ def upload_to_path(instance, filename, folder):
...
+    :param time
+    :return: string object in "time ago" format
+    """
+    now = datetime.utcnow().replace(tzinfo=utc)
+    diff = now - time
+    second_diff = diff.seconds
+    day_diff = diff.days
+
+    if day_diff == 0:
```
Comment (saad-waqar) [inline openedx/features/idea/helpers.py:31]: https://docs.djangoproject.com/en/3.0/ref/contrib/humanize/

Comment (saad-waqar) [reply openedx/features/idea/helpers.py:31]: https://github.com/philanthropy-u/edx-platform/blob/cc2ca35413b18f56e55fd5a14a21ce07c6dd75c2/openedx/features/teams/serializers.py#L96

File: openedx/features/idea/tests/views/test_idea_list_view.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+    @classmethod
+    def setUpClass(cls):
+        super(IdeaListingViewTest, cls).setUpClass()
+        configure_philu_theme()
+
+    def test_idea_list_view(self):
+        response = self.client.get(reverse('idea-listing'))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(IdeaListingView.paginate_by, 9)
```
Comment (ahmed-faraz46) [inline openedx/features/idea/tests/views/test_idea_list_view.py:19]: Please let's assert it as we haven't customized pagination functionality of Django.

File: openedx/features/idea/tests/views/test_idea_list_view.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+        self.assertEqual(IdeaListingView.ordering, ['-created'])
```
Comment (ahmed-faraz46) [inline openedx/features/idea/tests/views/test_idea_list_view.py:20]: same is the case with this one.

File: openedx/features/idea/helpers.py (hunk 3 above)
Comment (abdullah998) [reply openedx/features/idea/helpers.py]: changed it bro

File: openedx/features/idea/tests/views/test_idea_list_view.py (hunk 4 above)
Comment (abdullah998) [reply openedx/features/idea/tests/views/test_idea_list_view.py:19]: Removed it bro

File: openedx/features/idea/tests/views/test_idea_list_view.py (hunk 5 above)
Comment (abdullah998) [reply openedx/features/idea/tests/views/test_idea_list_view.py]: Removed it Ahmed bhai.

File: openedx/features/idea/tests/views/test_idea_list_view.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+from django.test import TestCase
+from django.urls import reverse
+from rest_framework import status
+
+from openedx.core.djangolib.testing.philu_utils import configure_philu_theme
```
Comment (zeshanarifios) [inline openedx/features/idea/tests/views/test_idea_list_view.py:5]: run isort please

File: openedx/features/idea/models.py [hunk 7]
```diff
@@ -5,6 +5,7 @@
 from django.contrib.auth.models import User
 from django.core.validators import FileExtensionValidator
 from django_countries.fields import CountryField
+from model_utils.models import TimeStampedModel
```
Comment (zeshanarifios) [inline openedx/features/idea/models.py:8]: run isort please

File: openedx/features/idea/views.py [hunk 8]
```diff
@@ -12,11 +15,13 @@ def get(self, request, *args, **kwargs):
         return render_to_response(self.template_name, {})
 
 
-class IdeaListingView(View):
+class IdeaListingView(ListView):
+    model = Idea
+    context_object_name = 'idea_list'
+    paginate_by = 9
```
[the remark this answers is not in the document]
Comment (abdullah998) [reply openedx/features/idea/views.py:21]: Yes Zeshan bhai

File: openedx/features/idea/models.py (hunk 7 above)
Comment (abdullah998) [reply openedx/features/idea/models.py:11]: Done Zeshan bhai.

File: openedx/features/idea/tests/views/test_idea_list_view.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+from openedx.features.idea.views import IdeaListingView
+
```
[the remark this answers is not in the document]
Comment (abdullah998) [reply openedx/features/idea/tests/views/test_idea_list_view.py]: Done Zeshan bhai.

**Pull request merged**